From 8ff31fe9aaa6a28ae84797de88028f600cff2e56 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 6 Jun 2023 11:03:47 +0200
Subject: [PATCH 3/4] configure: don't force the _WIN32_WINNT if it's high
 enough

Downgrading the value hides some API that can be useful in UWP builds.
---
 configure.ac | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c7016344..25deb7ba 100644
--- a/configure.ac
+++ b/configure.ac
@@ -38,7 +38,6 @@ case "${host_os}" in
         ;;
       *mingw*)
         SYS=mingw32
-        AC_DEFINE([_WIN32_WINNT], 0x0502, [Define to '0x0502' for Windows XP SP2 APIs.])
         AC_DEFINE([_WIN32_IE], 0x0501, [Define to '0x0501' for IE 5.01.])
         CPPFLAGS="${CPPFLAGS} -D__USE_MINGW_ANSI_STDIO=1"
         ;;
@@ -69,6 +68,23 @@ case "${host_os}" in
     ;;
 esac
 
+dnl Only force the _WIN32_WINNT if it's not set or lower than Windows XP SP2
+AS_IF([test "$SYS" = mingw32],[
+    AC_PREPROC_IFELSE([AC_LANG_PROGRAM(
+        [[#ifdef _WIN32_WINNT
+        # error _WIN32_WINNT already defined
+        #else
+        # include <sdkddkver.h>
+        # if defined(_WIN32_WINNT) && _WIN32_WINNT >= 0x0502
+        #  error _WIN32_WINNT toolchain default high enough
+        # endif
+        #endif
+        ]],[[;]])
+    ],[
+        AC_DEFINE([_WIN32_WINNT], [0x0502], [Define to '0x0502' for Windows XP SP2 APIs.])
+    ])
+])
+
 dnl Falsely enables for NetBSD
 dnl warning: warning: reference to obsolete getfsstat(); use getvfsstat()
 AS_IF([test "${SYS}" != "netbsd"], [
-- 
2.37.3.windows.1

